src/olympia/versions/models.py

```diff
@@ -896,6 +904,14 @@ class VersionReviewerFlags(ModelBase):
     pending_rejection = models.DateTimeField(
         default=None, null=True, blank=True, db_index=True
     )
+    pending_rejection_by = models.ForeignKey(
+        UserProfile, null=True, on_delete=models.CASCADE
+    )
+
+    def save(self, *args, **kwargs):
+        if not self.pending_rejection:
+            self.pending_rejection_by = None
```

Comment:
Ugh, I wish we had conditional database constraints so we could enforce "if pending_rejection is null, then pending_rejection_by has to be null", but mysql 😞